### ISSUE TITLE
[Feature Request] Destination Drive

### ISSUE THREAD

**Issue by BitSec01** [madMAx43v3r/chia-plotter#13]
Hello, Thanks for this amazing plotter, I tested it on my machine and it works very well (3800 seconds for 1 plot)
I was wondering if you could add an argument so that I can specify a destination drive where the final plot will be copied to.
This would make it easier for me to try and automate it on my plotting machine.

**Comment by RamonRobben** [comment]
My understanding would be that currently tmp_dir would be used as your final destination and tmp_dir2 as your nvme-ssd / ramdisk. Which would be fine. But I see that the first tmp_dir still needs to handle 25% of all writes. Can an HDD do this without bottlenecking? Or is it recommended to use like nvme/ram disk for tmp_dir2 and then use ssd for tmp_dir and move plots to your HDD's once they are done?

**Comment by madMAx43v3r** [comment]
> Or is it recommended to use like nvme/ram disk for tmp_dir2 and then use ssd for tmp_dir and move plots to your HDD's once they are done?
=> yes

`tmp_dir` still needs to be fast for phase 2 reading + writing and phase 3 reading. phase 1 writing to `tmp_dir` is not very much.

Copy to destination will be the next feature yes.

**Comment by RamonRobben** [comment]
@madMAx43v3r Ah alright so I understood it correctly. I wish I knew C and so I could contribute but unfortunately I am a complete noob at it so I'll leave it to you :)

**Comment by Dartellum** [comment]
> > Or is it recommended to use like nvme/ram disk for tmp_dir2 and then use ssd for tmp_dir and move plots to your HDD's once they are done?
> 
> => yes
> 
> `tmp_dir` still needs to be fast for phase 2 reading + writing and phase 3 reading. phase 1 writing to `tmp_dir` is not very much.
> 
> Copy to destination will be the next feature yes.

Will you use rsync to copy the file?

**Comment by madMAx43v3r** [comment]
No, I planned to run a thread that copies the data via reading + writing, parallel to running phase 1 of the next plot (if you do more than one).

**Comment by madMAx43v3r** [comment]
If you need rsync, you can simply do:
```
chia_plot ...
rsync *.plot destination/ && rm *.plot
```

**Comment by mist0706** [comment]
or do rsync plot*.plot /path/to/destination -avh --remove-source-files :)

**Comment by Dartellum** [comment]
> No, I planned to run a thread that copies the data via reading + writing, parallel to running phase 1 of the next plot (if you do more than one).

I was just curious. :smile:

**Comment by mist0706** [comment]
How do we define final directory with the new release?

**Comment by MarcusArthur** [comment]
chia_plot <pool_key> <farmer_key> [tmp_dir] [tmp_dir2] [final directory] [num_threads] [log_num_buckets]

**Comment by mist0706** [comment]
Looks like there is a bug in the final copy branch
![image](https://user-images.githubusercontent.com/10143260/121302638-48947480-c8fa-11eb-8e49-12f8289e8328.png)

Looking at the final destination folder it's already been copied successfully. It just tries to do it multiple times for no apparent reason?

**Comment by madMAx43v3r** [comment]
fixed and merged

**Comment by iw777** [comment]
i use `:~$ rsync -avh --info=progress *.plot /mnt/pfolder/ --remove-source-files` in linux to move final .plot to my main farmer, and so far its working fine, but I have to run it manually, I want to make it permanent listening? similar to `@echo off
:loop
set "source=C:\PLOT"
set "destination=Z:\PLOT"
robocopy "%source%" "%destination%" /mov *.plot
timeout /t 300
goto loop` in windows?

Linux >
<img width="614" alt="Screen Shot 2021-06-16 at 6 14 18 PM" src="https://user-images.githubusercontent.com/85085939/122301526-c9410b00-cece-11eb-84eb-0761f5765f52.png">

Windows>

<img width="767" alt="Screen Shot 2021-06-16 at 6 14 43 PM" src="https://user-images.githubusercontent.com/85085939/122301551-d0681900-cece-11eb-8140-9a2d1f74ccbb.png">

**Comment by mist0706** [comment]
Just add "while true; do rsync ......; sleep 60; done"

**Comment by iw777** [comment]
> Just add "while true; do rsync ......; sleep 60; done"

not sure what i am doing wrong. :(

this is my scrip >
<img width="803" alt="Screen Shot 2021-06-18 at 2 11 54 PM" src="https://user-images.githubusercontent.com/85085939/122602019-26a89980-d040-11eb-9183-3c7df6569f4a.png">

the error >
<img width="584" alt="Screen Shot 2021-06-18 at 2 12 46 PM" src="https://user-images.githubusercontent.com/85085939/122602044-2f996b00-d040-11eb-88f6-bf338a210a10.png">
